src/routes/communityGroups/community_groups.contoller.js

```diff
@@ -0,0 +1,82 @@
+const fs = require("fs");
+const { promisify } = require("util");
+
+const { ERROR_CODES } = require("../../middleware/error-handler");
+
+const data = require("../../../db/communityGtoups.data.json");
+
+const writeFile = promisify(fs.writeFile);
+
+const dbPath = "db/communityGtoups.data.json";
+
+// return all community groups
+const listCommunityGroups = (req, res) => {
+  return res.status(200).json(data);
+};
+
+// get a community group by its id if it exists
+const getCommunityGroup = (req, res) => {
+  const record = data.filter(
+    record => record.id === parseInt(req.params.id)
+  )[0];
+
+  if (!record) {
+    throw new Error(
+      JSON.stringify({
+        status: ERROR_CODES.NOT_FOUND,
+        message: "Record not found"
+      })
+    );
+  }
+
+  return res.status(200).json(record);
+};
+
+// add a community group if the request has no errors
+const addCommunityGroup = async (req, res) => {
+  const id = data.length + 1;
```

Comment:
What will happen if you delete some community groups and then add a new one? Will the id still be unique?

Reply:
No, it won't, but that wasn't a top concern, with data on json being only temporary until we integrate DB. If by chance we get duplicate ids, getting, updating and deleting them should account for this scenario:

**getting**
Returns the first with that id
`data.filter(    record => record.id === parseInt(req.params.id))[0]`

**updating and deleting**
Affect all with that id
` if (group.id === parseInt(req.params.id))`

Reply:
Very good!